http/client.go

```diff
@@ -58,7 +58,7 @@ func (c *Client) NewTxn(ctx context.Context, readOnly bool) (client.Txn, error)
 	methodURL := c.http.apiURL.JoinPath("tx")
 	methodURL.RawQuery = query.Encode()
 
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, methodURL.String(), nil)
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodPost, methodURL.String(), nil)
```

Comment:
thought: You merged this already but this is a bad idea as it no longer provided a means for the caller to set a timeout out to otherwise cancel the request that could hang indefinitely.

Reply:
Maybe we can add a http-only overload for this, I really don't think the internal and embedded code, and anything built on it should have to worry about context here.

Reply:
> I really don't think the internal and embedded code, and anything built on it should have to worry about context here.

I agree. But not having a better solution for it yet was the reason I didn't previously remove the context from those methods. Would love to find a better alternative.